ops/testing.py

```diff
@@ -397,14 +452,14 @@ def _get_config(self, charm_config: typing.Optional[dict]):
                 charm_config = '{}'
         elif isinstance(charm_config, str):
             charm_config = dedent(charm_config)
-        charm_config = yaml.safe_load(charm_config)
+        config = cast('RawConfig', yaml.safe_load(cast(str, charm_config)))
 
-        if not isinstance(charm_config, dict):
-            raise TypeError(charm_config)
-        return charm_config
+        if not isinstance(config, dict):  # pyright: reportUnnecessaryIsInstance=false
+            raise TypeError(config)
+        return config
```

Comment:
I think we could clean up these casts with something like (a diff applied after your diff):

```diff
-        config = cast('RawConfig', yaml.safe_load(cast(str, charm_config)))
+        config = yaml.safe_load(cast(str, charm_config))
 
-        if not isinstance(config, dict):  # pyright: reportUnnecessaryIsInstance=false
+        if not isinstance(config, dict):
             raise TypeError(config)
-        return config
+        return cast(RawConfig, config)
```